hermes/src/store.rs

```diff
@@ -288,14 +295,365 @@ impl Store {
+                .collect(),
+            magic: [0; 4],
+            ring_size: 100,
+        };
+        updates.push(Update::AccumulatorMessages(accumulator_messages.clone()));
+
+        // Wormhole merkle update
+        let merkle_tree = MerkleTree::<Keccak160>::from_set(
+            accumulator_messages.raw_messages.iter().map(|m| m.as_ref()),
+        )
+        .unwrap();
+
+        let wormhole_message = WormholeMessage::new(WormholePayload::Merkle(WormholeMerkleRoot {
+            slot,
+            ring_size: 100,
+            root: merkle_tree.root.as_bytes().try_into().unwrap(),
+        }));
+
+        let wormhole_message =
+            pythnet_sdk::wire::to_vec::<_, byteorder::BE>(&wormhole_message).unwrap();
+
+        let vaa = Vaa {
+            nonce: 0,
+            version: 0,
+            sequence,
+            timestamp: 0,
+            signatures: vec![],    // We are bypassing signature check now
+            guardian_set_index: 0, // We are bypassing signature check now
+            emitter_chain: Chain::Pythnet,
+            emitter_address: Address(pythnet_sdk::ACCUMULATOR_EMITTER_ADDRESS),
+            consistency_level: 0,
+            payload: serde_wormhole::RawMessage::new(wormhole_message.as_ref()),
+        };
+
+        updates.push(Update::Vaa(serde_wormhole::to_vec(&vaa).unwrap()));
+
+        updates
+    }
+
+    /// Create a dummy price feed base on the given seed for all the fields except
+    /// `publish_time` and `prev_publish_time`. Those are set to the given value.
+    pub fn create_dummy_price_feed_message(seed: u8, publish_time: i64) -> PriceFeedMessage {
+        PriceFeedMessage {
+            feed_id: [seed; 32],
+            price: seed as _,
+            conf: seed as _,
+            exponent: 0,
+            ema_conf: seed as _,
+            ema_price: seed as _,
+            publish_time,
+            prev_publish_time: publish_time,
+        }
+    }
+
+    pub async fn setup_store(cache_size: u64) -> (Arc<Store>, Receiver<()>) {
+        let (update_tx, update_rx) = tokio::sync::mpsc::channel(1000);
+        let store = Store::new(update_tx, cache_size);
+
+        // Add an initial guardian set with public key 0
+        store
+            .update_guardian_set(
+                0,
+                GuardianSet {
+                    keys: vec![[0; 20]],
+                },
+            )
+            .await;
+
+        (store, update_rx)
+    }
+
+    #[tokio::test]
+    pub async fn test_store_works() {
+        let (store, mut update_rx) = setup_store(10).await;
+
+        let price_feed_message = create_dummy_price_feed_message(100, 10);
+        // Populate the store with some inputs via calling them all at once and joining them.
+        let res = join_all(
+            generate_update(vec![Message::PriceFeedMessage(price_feed_message)], 10, 20)
+                .into_iter()
+                .map(|u| store.store_update(u)),
+        )
+        .await;
```

Comment:
maybe make a store_updates helper so you can do something like `store_updates(generate_updates(...)).await`

you could actually put the assert below into the helper, as that seems like it's a sanity check not part of the test logic (?)

Reply:
👍